### ISSUE TITLE
unexpected keyword argument 'io_loop'

### ISSUE THREAD

**Issue by delubi** [saltstack/salt#52157]
### Description of Issue/Question
I installed Salt on Ubuntu18.04 (running as a VM inside VirtualBox). Getting error below when doing salt-run.

root@ubuntu:~# salt-run http.query google.com
[ERROR   ] Future exception was never retrieved
future: <Future finished exception=TypeError("__init__() got an unexpected keyword argument 'io_loop'",)>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 326, in wrapper
    yielded = next(result)
  File "/usr/local/lib/python3.6/dist-packages/salt/transport/ipc.py", line 335, in _connect
    io_loop=self.io_loop,
  File "/usr/local/lib/python3.6/dist-packages/tornado/iostream.py", line 1222, in __init__
    super(IOStream, self).__init__(*args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'io_loop'

### Setup
No special config or state file. Just simple salt-run after default install.

### Steps to Reproduce Issue
root@ubuntu:~# salt-run -l debug http.query google.com
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: ubuntu
[DEBUG   ] Missing configuration file: /root/.saltrc
[DEBUG   ] Configuration file path: /etc/salt/master
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] LazyLoaded http.query
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: ubuntu
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Please install 'virt-what' to improve results of the 'virtual' grain.
[DEBUG   ] Using selector: EpollSelector
[DEBUG   ] Using selector: EpollSelector
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Popen(['git', 'version'], cwd=/root, universal_newlines=False, shell=None)
[DEBUG   ] Popen(['git', 'version'], cwd=/root, universal_newlines=False, shell=None)
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pull.ipc
[ERROR   ] Future exception was never retrieved
future: <Future finished exception=TypeError("__init__() got an unexpected keyword argument 'io_loop'",)>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 326, in wrapper
    yielded = next(result)
  File "/usr/local/lib/python3.6/dist-packages/salt/transport/ipc.py", line 335, in _connect
    io_loop=self.io_loop,
  File "/usr/local/lib/python3.6/dist-packages/tornado/iostream.py", line 1222, in __init__
    super(IOStream, self).__init__(*args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'io_loop'

### Versions Report
root@ubuntu:~# salt --versions-report
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: unknown
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: 2.1.11
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.26.0
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.2
         Python: 3.6.7 (default, Oct 22 2018, 11:32:17)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 17.1.2
           RAET: Not Installed
          smmap: 2.0.5
        timelib: 0.2.4
        Tornado: 5.1.1
            ZMQ: 4.2.5

System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-46-generic
         system: Linux
        version: Ubuntu 18.04 bionic

**Comment by Ch3LL** [comment]
you are running into this because currently tornado 5.0 on python3 is not supported by salt. You can either downgrade tornado to <5.0 or use python2 with salt.

**Comment by delubi** [comment]
@Ch3LL Thanks! I downgraded tornado as below. 

python3 -m pip install 'tornado<5.0' --force-reinstall

After that the io_loop error was gone. However, I got another error as:

msgpack.exceptions.UnpackValueError: 'utf-8' codec can't decode byte 0x82 in position 22: invalid start byte

Ended up uninstalling salt and then ran pip3 install salt which installed latest version 2019.2.0.

Also updated the /lib/systemd/system/salt-master.service and salt-minion.service to use the salt-master/minion in /usr/local/bin (python3 version) as opposed to /usr/bin (python2 version).

Now it's working as expected.
